### PR TITLE
refactor(batch): removing db fields in batch encoding type

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

To let the code style be consistent with `InsertChunk`.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

File: coordinator/internal/orm/batch.go [hunk 1]
```diff
@@ -190,21 +190,69 @@ func (o *Batch) InsertBatch(ctx context.Context, batch *encoding.Batch, dbTX ...
...
+	// if parentBatch==nil then err==gorm.ErrRecordNotFound, which means there's
+	// no batch record in the db, we then use default empty values for the creating batch;
+	// if parentBatch!=nil then err==nil, then we fill the parentBatch-related data into the creating batch
+	if parentBatch != nil {
+		startChunkIndex = parentBatch.EndChunkIndex + 1
+	}
+
+	startDAChunk, err := codecv0.NewDAChunk(batch.Chunks[0], batch.TotalL1MessagePoppedBefore)
+	if err != nil {
+		log.Error("failed to new start DA chunk", "index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
```
Comment (Thegaram) [inline coordinator/internal/orm/batch.go]:
```suggestion
		log.Error("failed to create new start DA chunk", "index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
```

File: coordinator/internal/orm/batch.go [hunk 2]
```diff
@@ -190,21 +190,69 @@ func (o *Batch) InsertBatch(ctx context.Context, batch *encoding.Batch, dbTX ...
...
+	}
+
+	startDAChunkHash, err := startDAChunk.Hash()
+	if err != nil {
+		log.Error("failed to get start DA chunk hash", "index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
+			"parent hash", batch.ParentBatchHash, "number of chunks", numChunks, "err", err)
+		return nil, fmt.Errorf("Batch.InsertBatch error: %w", err)
+	}
+
+	endDAChunk, err := codecv0.NewDAChunk(batch.Chunks[numChunks-1], batch.TotalL1MessagePoppedBefore)
```
Comment (Thegaram) [inline coordinator/internal/orm/batch.go]: Using `batch.TotalL1MessagePoppedBefore` here is incorrect, you need to calculate `TotalL1MessagePoppedBeforeChunk`.

Comment (colinlyguo) [reply coordinator/internal/orm/batch.go]: fixed in [a42f810](https://github.com/scroll-tech/scroll/pull/1196/commits/a42f8106b8eedbb512691def98119cad5f7f9fa3)

File: coordinator/internal/orm/batch.go (hunk 1 above)
Comment (colinlyguo) [reply coordinator/internal/orm/batch.go]: changed in [a42f810](https://github.com/scroll-tech/scroll/pull/1196/commits/a42f8106b8eedbb512691def98119cad5f7f9fa3)

File: coordinator/internal/orm/batch.go [hunk 3]
```diff
@@ -190,21 +190,73 @@ func (o *Batch) InsertBatch(ctx context.Context, batch *encoding.Batch, dbTX ...
...
 		log.Error("failed to create new DA batch",
 			"index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
-			"parent hash", batch.ParentBatchHash, "number of chunks", len(batch.Chunks), "err", err)
+			"parent hash", batch.ParentBatchHash, "number of chunks", numChunks, "err", err)
 		return nil, err
 	}
 
+	var startChunkIndex uint64
+	parentBatch, err := o.GetLatestBatch(ctx)
+	if err != nil && !errors.Is(errors.Unwrap(err), gorm.ErrRecordNotFound) {
```
Comment (Thegaram) [inline coordinator/internal/orm/batch.go]: Do we need to unwrap here?

**Review by Thegaram — DISMISSED** [review]

File: coordinator/internal/orm/batch.go (hunk 3 above)
Comment (colinlyguo) [reply coordinator/internal/orm/batch.go]: `GetLatestBatch` would not return `gorm.ErrRecordNotFound` error:
https://github.com/scroll-tech/scroll/blob/a42f8106b8eedbb512691def98119cad5f7f9fa3/rollup/internal/orm/batch.go#L139-L141

thus removed `errors.Is(errors.Unwrap(err), gorm.ErrRecordNotFound)` check.

also changed `getLatestChunk` and `InsertChunk`'s error handling.

related commit: https://github.com/scroll-tech/scroll/pull/1196/commits/d8ae98707fad2af4b5e5986fc214ae48776b2734

**Review by Thegaram — APPROVED** [review]

File: coordinator/internal/orm/batch.go [hunk 4]
```diff
@@ -190,21 +193,73 @@ func (o *Batch) InsertBatch(ctx context.Context, batch *encoding.Batch, dbTX ...
...
+		log.Error("failed to get latest batch", "index", batch.Index, "total l1 message popped before", batch.TotalL1MessagePoppedBefore,
+			"parent hash", batch.ParentBatchHash, "number of chunks", numChunks, "err", err)
+		return nil, fmt.Errorf("Batch.InsertBatch error: %w", err)
+	}
+
+	// if parentBatch==nil then err==gorm.ErrRecordNotFound, which means there's
+	// no batch record in the db, we then use default empty values for the creating batch;
+	// if parentBatch!=nil then err==nil, then we fill the parentBatch-related data into the creating batch
+	if parentBatch != nil {
+		startChunkIndex = parentBatch.EndChunkIndex + 1
```
Comment (georgehao) [inline coordinator/internal/orm/batch.go:221]: Isn't the Batch.index calculate here?

Comment (Thegaram) [reply coordinator/internal/orm/batch.go:221]: No, batch index is already known (`batch.Index`).

But `orm.Batch.StartChunkIndex` is not contained in `batch` anymore. So here we recalculate it, using the parent batch. There should be no gaps between the parent and the current batch, so `currentBatch.startChunkIndex = parentBatch.EndChunkIndex + 1`.

**Review by 0xmountaintop — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**